apps/passport-server/src/database/queries/e2ee.ts

```diff
@@ -64,10 +65,10 @@ export async function updateEncryptedStorage(
       ($1, $2) ON CONFLICT(blob_key) DO UPDATE SET encrypted_blob = $1`,
       [newBlobKey, encryptedBlob]
     );
-    await txClient.query("UPDATE users SET salt = $2 WHERE uuid = $1", [
-      uuid,
-      newSalt
-    ]);
+    await txClient.query(
+      "UPDATE users SET salt = $2, encryption_key = NULL WHERE uuid = $1",
+      [uuid, newSalt]
+    );
```

Comment:
this overlaps with @artwyman's PR. recommend coordinating regarding who merges first.